monai/networks/nets/daf3d.py

```diff
@@ -0,0 +1,574 @@
+                kernel_size=1,
+                adn_ordering="NA",
+                norm=group_norm,
+                act=act_prelu,
+            ),
+            Convolution(
+                spatial_dims=3,
+                in_channels=64,
+                out_channels=64,
+                kernel_size=3,
+                adn_ordering="NA",
+                padding=1,
+                norm=group_norm,
+                act=act_prelu,
+            ),
+            Convolution(
+                spatial_dims=3,
+                in_channels=64,
+                out_channels=64,
+                kernel_size=3,
+                adn_ordering="NA",
+                padding=1,
+                norm=group_norm,
+                act=act_prelu,
+            ),
+        )
+        self.attention = AttentionModule(
+            spatial_dims=3, in_channels=192, out_channels=64, norm=group_norm, act=act_prelu
+        )
+
+        self.refine = Convolution(3, 256, 64, kernel_size=1, adn_ordering="NA", norm=group_norm, act=act_prelu)
+        self.predict2 = nn.Conv3d(64, out_channels, kernel_size=1)
+        self.aspp = Daf3dASPP(
+            spatial_dims=3,
+            in_channels=64,
+            conv_out_channels=64,
+            out_channels=64,
+            kernel_sizes=(3, 3, 3, 3),
+            dilations=((1, 1, 1), (1, 6, 6), (1, 12, 12), (1, 18, 18)),  # type: ignore
+            norm_type=group_norm,
+            acti_type=None,
+            bias=True,
+        )
+
+    def forward(self, x):
+        # layers from 1 - 4
+        single_layer_features = list(self.backbone_with_fpn(x).values())
+
+        # first 4 supervised signals (SLFs 1 - 4)
+        supervised1 = [self.predict1(slf) for slf in single_layer_features]
+
+        mlf = self.fuse(torch.cat(single_layer_features, 1))
+
+        attentive_features_maps = [self.attention(slf, mlf) for slf in single_layer_features]
+        att_features, att_maps = tuple(zip(*attentive_features_maps))
+
+        # second 4 supervised signals (af 1 - 4)
+        supervised2 = [self.predict2(af) for af in att_features]
+
+        # attentive maps as optional additional output
+        supervised3 = [self.predict2(am) for am in att_maps]
+
+        attentive_mlf = self.refine(torch.cat(att_features, 1))
+
+        aspp = self.aspp(attentive_mlf)
+
+        supervised_final = self.predict2(aspp)
+
+        if self.training:
+            output = supervised1 + supervised2 + [supervised_final]
+            output = [F.interpolate(o, size=x.size()[2:], mode="trilinear") for o in output]
+        else:
+            if self.visual_output:
+                supervised_final = F.interpolate(supervised_final, size=x.size()[2:], mode="trilinear")
+                supervised_inner = [
+                    F.interpolate(o, size=x.size()[2:], mode="trilinear")
+                    for o in supervised1 + supervised2 + supervised3
+                ]
+                output = supervised_final + supervised_inner
```

Comment:
btw `supervised_final` is a tensor and `supervised_inner` is a list, it looks like a bug adding them together...

Reply:
Yes you're right I forgot the list brackets